src/common/devcard.ts

```diff
@@ -46,15 +52,38 @@ const getFavoriteSourcesLogos = async (
     )
     .innerJoin(
       Source,
-      'source',
-      'source.id = p."sourceId" and source.active = true and source.private = false',
+      alias,
+      `"${alias}".id = p."sourceId" and "${alias}".active = true and "${alias}".private = false`,
     )
     .where('v."userId" = :id', { id: userId })
     .andWhere(`s.count > 10`)
     .groupBy('p."sourceId"')
     .orderBy('count(*) * 1.0 / min(s.count)', 'DESC')
-    .limit(5)
-    .getRawMany();
+    .limit(limit);
+};
+
+export const getFavoriteSourcesIds = async (
+  con: DataSource,
+  userId: string,
+  limit = 5,
+): Promise<string[]> => {
+  const query = generateFavoriteSourcesQuery(con, userId, { limit });
+  const sources = await query.getRawMany();
+
+  return sources.map((source) => source.id);
+};
+
+export const getFavoriteSourcesLogos = async (
+  con: DataSource,
+  userId: string,
+): Promise<string[]> => {
+  const alias = 'source';
+  const query = generateFavoriteSourcesQuery(con, userId, {
+    alias,
+    limit: 5,
+  }).addSelect(`min("${alias}".image)`, 'image');
+  const sources = await query.getRawMany();
+
   return sources.map((source) => source.image);
 };
```

Comment:
Do we query these seperate or always same time?
(Feels like it's just one query with different graphQL props?